api/views.py

```diff
@@ -34,6 +34,12 @@ class QuestionCategoryViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = QuestionCategorySerializer
 
 
+class TokenValueViewSet(viewsets.ModelViewSet, mixins.UpdateModelMixin):
+    queryset = TokenValue.objects.all()
```

Comment:
Here you need to make sure that all the TokenValue objects exist to prevent getting Null for some category-difficulty pair.

Reply:
So after looking into the objects, it seems like there aren't any null values in general since when a Tokenvalue object is saved, it is always assigned a default value. There isn't any Null values in the Angular side anymore either. So should I still run a null check and just exclude the ones in the queryset that are null?